src/components/Sorting.jsx

```diff
@@ -0,0 +1,47 @@
+import * as React from "react";
+import InputLabel from "@mui/material/InputLabel";
+import MenuItem from "@mui/material/MenuItem";
+import FormControl from "@mui/material/FormControl";
+import Select from "@mui/material/Select";
+import { Link, useLocation } from "react-router-dom";
+
+const Sorting = () => {
+  const { pathname } = useLocation();
+  const [sortType, setSortType] = React.useState("");
+
+  const handleSelectChange = (event, newValue) => {
+    setSortType(event.target.value);
+  };
+  return (
+    <div>
+      <FormControl className="sort" sx={{ m: 1, minWidth: 120 }} size="small">
+        <InputLabel id="demo-select-small">Sort!</InputLabel>
+        <Select
+          labelId="demo-select-small"
+          id="demo-select-small"
+          value={sortType}
+          label="Sort!"
+          onChange={handleSelectChange}
+        >
+          <MenuItem value="asc">
+            <Link to={`${pathname}?order=asc`}>Ascending</Link>
+          </MenuItem>
+          <MenuItem value="desc">
+            <Link to={`${pathname}?order=desc`}>Descending</Link>
+          </MenuItem>
+          <MenuItem value="date">
+            <Link to={`${pathname}?sort=created_at`}>Date</Link>
+          </MenuItem>
+          <MenuItem value="comments">
+            <Link to={`${pathname}?sort=comment_count`}>Comment count</Link>
+          </MenuItem>
+          <MenuItem value="votes">
+            <Link to={`${pathname}?sort=votes`}>Votes</Link>
+          </MenuItem>
```

Comment:
Slightly weird bug here - if you click one of the options in the drop-down menu but you _don't_ click the text specifically, it updates the value of the `select` but it doesn't actually change the data, because the `Link` part of it is only wrapped around the text. You might have to put the text in a box which fills the space of each of the menu items inside your links, so that the whole thing is clickable.

On another note, I think it is a little confusing having the "ascending/descending" options in the same dropdown. The functionality is there but I think it would be a better UX to have two separate menus so that you can pick the sort option, and then pick the order, and you can still see both on screen.